custom/enikshay/tasks.py

```diff
@@ -185,14 +187,26 @@ def run_batch(self, case_ids):
                     noupdate_count += 1
                 if len(updates) >= batch_size:
                     bulk_update_cases(self.domain, updates, device_id)
+                    self._update_ledger_values(episode_case_ids)
+                    episode_case_ids = [] # reset episode case ids
                     updates = []
                     case_batches += 1
 
             if len(updates) > 0:
                 bulk_update_cases(self.domain, updates, device_id)
+            self._update_ledger_values(episode_case_ids)
 
         return BatchStatus(update_count, noupdate_count, success_count, errors, case_batches, t.interval)
 
+    def _update_ledger_values(self, episode_case_ids):
+        accessor = CaseAccessors(self.domain)
+        for episode_case_id in episode_case_ids:
+            # fetch episode case again to get updated case
+            episode_case = accessor.get_case(episode_case_id)
+            adherence_cases = get_adherence_cases_from_episode(episode_case)
```

Comment:
You can use `AdherenceDatastore` which is already initialized in this class (see, `adherence_data_store`)

Reply:
You will be able to avoid these calculations if you move this logic to `EpisodeAdherenceUpdate`, calculate ledger updates, and then commit them in bulk [here](https://github.com/dimagi/commcare-hq/pull/18896/files#diff-7ede24022ad113f7bd1a083192d23896R199)